akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/EventSourcedBehaviorTestKitImpl.scala

```diff
@@ -150,10 +132,48 @@ import akka.persistence.typed.internal.EventSourcedBehaviorImpl
       replyProbe.stop()
     }
 
+    val newState = getState()
+    val newEvents = getEvents(oldEvents.size)
+
+    postCommandCheck(newEvents, newState, Some(reply))
+
+    CommandResultImpl[Command, Event, State, R](command, newEvents, newState, Some(reply))
+  }
+
+  private def getEvents[R](dropOldEvents: Int): immutable.Seq[Event] = {
+    // FIXME we can expand the api of persistenceTestKit to read from storage from a seqNr instead
+    persistenceTestKit.persistedInStorage(persistenceId.id).map(_.asInstanceOf[Event]).drop(dropOldEvents)
```

Comment:
FYI there is an impl of `currentEventsByPersistenceId` in `PersistenceTestKitReadJournal`.